pool/helpers.go

```diff
@@ -50,6 +56,40 @@ func runJob(ctx context.Context, job amboy.Job, q amboy.Queue, startAt time.Time
 		defer cancel()
 	}
 
+	if err := job.Lock(q.ID()); err != nil {
+		job.AddError(errors.Wrap(err, "problem locking job"))
+		return false
+	}
+	if err := q.Save(ctx, job); err != nil {
+		job.AddError(errors.Wrap(err, "problem saving job state"))
+		return false
+	}
+
+	pingerCtx, stopPing := context.WithCancel(ctx)
```

Comment:
`defer stopPing()` instead of calling at the end of the function

Reply:
my instinct is that you want to be sure that this is not running before you call complete.

Reply:
but I think you can call cancel more than once so it's all good!